### DISCUSSION TITLE
Effect of automatic_scaling in thermal hydraulics

### DISCUSSION THREAD

**Discussion by jvwilliams23** [idaholab/moose#25921] [Q&A Modules: Thermal Hydraulics] [answered]
I am trying to fix a failing solve in my troublesome THM case by following the webpage ["Troubleshooting failed solves"](https://mooseframework.inl.gov/application_usage/failed_solves.html). I think it is an issue in variable scaling in my case, so I try to use `automatic_scaling = true` (which did not work). So using the [01_flow_channel.i](https://github.com/idaholab/moose/blob/next/modules/thermal_hydraulics/tutorials/single_phase_flow/01_flow_channel.i) tutorial, I make the same modification (`automatic_scaling=true`, as given below) which seems to crash at the second time step.

My questions are: is `automatic_scaling` not appropriate for THM problems? Why is this simple case not working with `automatic_scaling=true`? Also, in general, are the guidelines in "Troubleshooting failed solves" document suitable for THM problems? 

```
T_in = 300. # K
m_dot_in = 1e-2 # kg/s
press = 10e5 # Pa

[GlobalParams]
  initial_p = ${press}
  initial_vel = 0.0001
  initial_T = ${T_in}
  gravity_vector = '0 0 0'

  rdg_slope_reconstruction = minmod
  closures = thm_closures
  fp = he
[]

[FluidProperties]
  [he]
    type = IdealGasFluidProperties
    molar_mass = 4e-3
    gamma = 1.67
    k = 0.2556
    mu = 3.22639e-5
  []
[]

[Closures]
  [thm_closures]
    type = Closures1PhaseTHM
  []
[]

[Components]
  [inlet]
    type = InletMassFlowRateTemperature1Phase
    input = 'core_chan:in'
    m_dot = ${m_dot_in}
    T = ${T_in}
  []

  [core_chan]
    type = FlowChannel1Phase
    position = '0 0 0'
    orientation = '0 0 1'
    length = 1
    n_elems = 25
    A = 7.2548e-3
    D_h = 7.0636e-2
  []

  [outlet]
    type = Outlet1Phase
    input = 'core_chan:out'
    p = ${press}
  []
[]


[Preconditioning]
  [pc]
    type = SMP
    full = true
  []
[]

[Executioner]
  type = Transient
  solve_type = NEWTON
  line_search = basic

  start_time = 0
  end_time = 1000
  dt = 10

  petsc_options_iname = '-pc_type'
  petsc_options_value = 'lu'

  nl_rel_tol = 1e-8
  nl_abs_tol = 1e-8
  nl_max_its = 25

  automatic_scaling = true
[]

[Outputs]
  exodus = true

  print_linear_residuals = true
[]
[Debug]
  show_var_residual_norms = true
[]
```

**Comment by GiudGiud** [comment, marked as answer]
Hello

I dont think they use it much for THM, probably because the residuals for the energy equation are initialized closer to their solution so the solver tries to over-converge them. 
Also automatic scaling scales per variable and in TH simulations the friction factors difference in various components make it advantageous to scale per component

Instead they pass the scaling factors in the FlowChannel1Phase:
```
    scaling_factor_1phase = '1 1 1e-5'
```

Guillaume

**Reply by jvwilliams23** [reply]
Thanks!
